### PR TITLE
feat(block-section): add subcomponent CSS props

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #7180 

## Summary

Adds the following subcomponent tokens (CSS props):

* `--calcite-block-section-switch-corner-radius`
* `--calcite-block-section-switch-handle-background-color`
* `--calcite-block-section-switch-handle-border-color-hover`
* `--calcite-block-section-switch-handle-border-color`
* `--calcite-block-section-switch-handle-shadow-hover`
* `--calcite-block-section-switch-handle-shadow`
* `--calcite-block-section-switch-track-background-color-checked`
* `--calcite-block-section-switch-track-background-color`
* `--calcite-block-section-switch-track-border-color-checked`
* `--calcite-block-section-switch-track-border-color`

## Notes

* internal label does not require customization.
* icon-related CSS props will be added by https://github.com/Esri/calcite-design-system/pull/8868/.

### CODE REVIEW

File: packages/calcite-components/src/components/block-section/block-section.scss [hunk 1]
```diff
@@ -5,8 +5,18 @@
  *
  * @prop --calcite-block-section-background-color: Specifies the background color of the component.
  * @prop --calcite-block-section-border-color: Specifies the border color of the component. Applicable on bottom border when open.
- * @prop --calcite-block-section-heading-text-color: Specifies the component's heading text color.
  * @prop --calcite-block-section-heading-text-color-hover: Specifies the component's heading text color on hover.
+ * @prop --calcite-block-section-heading-text-color: Specifies the component's heading text color.
+ * @prop --calcite-block-section-switch-border-radius: Specifies the border radius of the switch.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/block-section/block-section.scss:10]: this should be "corner-radius"

**Review by alisonailea — CHANGES_REQUESTED** [review]
just a quick name change for corner radius

File: packages/calcite-components/src/components/block-section/block-section.scss (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.scss:10]: Oh, nice catch. Thanks! 🦅👁️

Comment (jcfranco) [reply packages/calcite-components/src/components/block-section/block-section.scss]: Updated!

**Review by alisonailea — APPROVED** [review]

**Pull request merged**